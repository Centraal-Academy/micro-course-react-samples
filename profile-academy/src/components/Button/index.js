import React from 'react'

export default function Button (props) {
  const { children, ...properties } = props
  return <button {...properties} >{ children }</button>
}
