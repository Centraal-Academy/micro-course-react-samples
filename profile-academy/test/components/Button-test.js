/* eslint-env jest */
/* global Enzyme */

import Button from '../../src/components/Button'
const { shallow } = Enzyme

describe('sample test with enzyme', () => {
  test('should have text content Something is missing when prop message dont exist', () => {
    const element = shallow(<Button>Click me!</Button>)
    expect(element).toMatchSnapshot()
    expect(element.text()).toEqual('Click me!')
  })
})
