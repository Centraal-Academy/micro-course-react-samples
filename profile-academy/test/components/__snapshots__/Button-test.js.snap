// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sample test with enzyme should have text content Something is missing when prop message dont exist 1`] = `
<button>
  Click me!
</button>
`;
